src/musicbrainz/web/musicbrainzrecordingstask.cpp

```diff
@@ -179,11 +179,17 @@ void MusicBrainzRecordingsTask::doNetworkReplyFinished(
     // Related Bug: https://bugs.launchpad.net/mixxx/+bug/1983204
     // In order to not hit the rate limits and respect their rate limiting rule.
     // We are going to delay every request by one second.
-
-    while (m_timer.elapsed(true) < kDelayDuration) {
+    if (m_timer.elapsed(true).toIntegerSeconds() >= 1) {
+        m_timer.restart(true);
+        slotStart(m_parentTimeoutMillis);
+        return;
+    } else {
+        auto sleepDuration = (kDelayDuration.toIntegerMillis() -
```

Comment:
You need to query the elapsed time only once.
This avoids that the timer elapses between the first and the second call.